DiagnosticsExtension/Models/ConnectionStringValidator/FileShareStorageValidator.cs

```diff
@@ -0,0 +1,117 @@
+﻿// -----------------------------------------------------------------------
+// <copyright file="StorageValidator.cs" company="Microsoft Corporation">
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+// </copyright>
+// -----------------------------------------------------------------------
+
+using DiagnosticsExtension.Controllers;
+using DiagnosticsExtension.Models.ConnectionStringValidator.Exceptions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Azure.Storage.Blobs;
+using Azure.Storage.Queues;
+using Azure.Storage.Files;
+using Azure.Storage.Files.Shares;
+using Microsoft.WindowsAzure.Storage;
+
+namespace DiagnosticsExtension.Models.ConnectionStringValidator
+{
+    public class FileShareStorageValidator : IConnectionStringValidator
+    {
+        public string ProviderName => "Microsoft.WindowsAzure.Storage";
+
+        public ConnectionStringType Type => ConnectionStringType.FileShareStorageAccount;
+        public async Task<ConnectionStringValidationResult> ValidateViaAppsettingAsync(string appsettingname, string entityName)
+        {
+            var response = new ConnectionStringValidationResult(Type);
+
+            try
+            {
+                var result = await TestConnectionStringViaAppSetting(appsettingname, entityName);
+                if (result.Succeeded)
+                {
+                    response.Status = ConnectionStringValidationResult.ResultStatus.Success;
+                }
+                else
+                {
+                    throw new Exception("Unexpected state reached: result.Succeeded == false is unexpected!");
+                }
+            }
+            catch (Exception e)
+            {
+                if (e is MalformedConnectionStringException)
```

Comment:
If the logic here is the same for all 3 storage validators (blob, queue, file) let's move the common code to a single utility function in a utility class?  This could make code maintenance easier and less error prone.